src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -289,8 +295,20 @@ public virtual FieldType AddField(FieldType fieldType)
             return builder;
         }
 
+        /// <summary>
+        /// Returns a builder for a new field.
+        /// </summary>
+        /// <typeparam name="TGraphType">The graph type of the field.</typeparam>
         public virtual FieldBuilder<TSourceType, object> Field<TGraphType>() => Field<TGraphType, object>();
 
+        /// <summary>
+        /// Returns a builder for a new field that is linked to a property of the source object.
+        /// </summary>
+        /// <typeparam name="TProperty">The return type of the field.</typeparam>
+        /// <param name="name">The name of this field.</param>
+        /// <param name="expression">The property of the source object represented within an expression.</param>
+        /// <param name="nullable">Indicates if this field should be nullable or not.</param>
```

Comment:
Please note in comments that this property is only used when the type property is not set (and for type property comment too). I have probably explained this behavior a hundred times already to anyone who encountered the incomprehensible behavior of their types.

Reply:
See #1604  and related issues.

Reply:
Do you really think we have to mention it twice?  See updated xml comments.

Reply:
It's not obligatory. At least it got better.